### ISSUE TITLE
Bot not clicking on fish

### ISSUE THREAD

**Issue by Laur1810** [vncsms/Metin2FishBot#12]
The bot doesn't click on the fish, it starts, put the bait, starts fishing but doesn't detect the fist in the circle in order to click on it, or maybe it just doesn't move the mouse and click it, the mouse just stands in the middle of the screen.
Also, during the "Since you activated the virtualenv you can install all requirements: `pip install -r requirements.txt`. Pip will install these following libraries:" step, the pyinstaller==4.3 dependency gives and error, so you need to install the latest version of it manually (either type them manually and install the latest version of pyinstaller or just edit the requirements.txt file before so that it doesn't have the ==4.3). #

**Comment by vncsms** [comment]
I made an update to the code. I removed all requirements versions, and updated the python version too.

**Comment by Laur1810** [comment]
Thanks! The installation part went a lot more smoothly now, but it seems that I have the same problem where the bot does not click on the fish at all, it puts the bait and fish but just doesn't click on the fish. I've attached a video if it helps:
https://drive.google.com/file/d/11TujHV8E4VByre2udjjuofYAeYgQ_izD/view?usp=share_link

**Comment by vncsms** [comment]
Try to run the cmd as admin, or the executable file, and check the name of yourt metin2 window game. My is configured to "metin2"

**Comment by Laur1810** [comment]
Yea, i did run the hack.py exec as admin and changed the constants.py file var from METIN2 to "World of Metin2" before running the pyinstall --onfile to create the dist folder but still the same

**Comment by vncsms** [comment]
I know the problem. Your serve set the fishing game o the position 0x0

**Comment by Laur1810** [comment]
Ah, yea, i thought of it as well, any way that you think I can fix this?

**Comment by vncsms** [comment]
Go to the file fishingbot.py. Remove the comment. DOnt delete the code just ''''
![image](https://github.com/vncsms/Metin2FishBot/assets/12662329/12ef5d4c-25a4-40c1-b156-76b1b0bd7652)

And then try to change these values:

![image](https://github.com/vncsms/Metin2FishBot/assets/12662329/e02d7a3a-565d-41f9-af72-3da6aa9e64f5)

**Comment by vncsms** [comment]
Try to change the position to 0

**Comment by Laur1810** [comment]
Unfortunately, still the same after changing  FISH_WINDOW_POSITION = (0, 0) to 0, 0, the cursor is located over the fishing game now but doesn't click the fish

**Comment by Laur1810** [comment]
also, i've re-run pyinstaller --onefile hack.py after making any changes to make sure

**Comment by vncsms** [comment]
You need to match the window, it's not 0,0 perfect.

**Comment by vncsms** [comment]
It will be easir if you run 'python hack.py' instead.

**Comment by Laur1810** [comment]
any tips on how I can find the perfect position on the window?

**Comment by Laur1810** [comment]
i managed to get it to work, I had to resize FISH_WINDOW_SIZE = (285, 260) to 285, 260 as well since the metin change it as well.
Thank you for your help glad that it works :D

**Comment by ghost** [comment]
U change just the FISH_WINDOW_SIZE to (285,260) or also the FISH_WINDOW_POSITION  ?

**Comment by ghost** [comment]
i have the kind isue, they moved the minigame to the top left in the corner

**Comment by Laur1810** [comment]
@revertl0l You need to change FISH_WINDOW_SIZE to the size that the minigame has (you can easily get it with a screenshot tool or smth like that) and since the metin that I played had the fishing game in the top left I also changed FISH_WINDOW_POSITION to (0, 0)

**Comment by ghost** [comment]
@Laur1810, i have to change first the values and after install the fishbot with cmd ?

**Comment by vncsms** [comment]
The changes are in the code, so if you wanna to run via executable you need to "recompile the code". If you are running via terminal, you just need to run the code.

**Comment by ghost** [comment]
Yea, i executable, so i will compile again, but i have 1 more question, how i can find the FISH_WINDOW_SIZE ?

**Comment by vncsms** [comment]
Look for the old messages. There's one explaining. https://github.com/vncsms/Metin2FishBot/issues/12#issuecomment-1541197899

**Comment by ghost** [comment]
No,no, i know how to change them, i ask how i can find the perfect values for me

**Comment by vncsms** [comment]
Try to remove the comment, as described in this post. When youo run the code a window is going to show up. You need to center the mini game in the window.

**Comment by ghost** [comment]
Yea, so the mouse it s on the mini game, but not in te circle now, and doen t works, have to put FISH_WINDOW_SIZE value perfectly, doent work with ur value laur, and i try it on wom2

**Comment by Laur1810** [comment]
@revertl0l I did it on WOM2 as well, all that I had to change is first in the constants file `GAME_NAME = "World of Metin2"`
Then, inside the fishingbot file, I changed the values to this:
```
    FISH_WINDOW_SIZE = (285, 260)
    FISH_WINDOW_POSITION = (0, 0)
```

That's all I had to do to make it work, make sure that you first change those values and only after compile the .exe by running `pyinstaller --onefile hack.py`

**Comment by ghost** [comment]
@Laur1810  managed to make it work in the end, but I had to set FISH_WINDOW_SIZE = (350, 340) it seems. However, I noticed some issues. Now, I'm not sure if it's just me because the fish window size isn't perfectly configured, but on average, out of 10 mini-games, the hack has a success rate of 60% in catching those 3 fish. The rest of the time, it only catches 2 fish and misses the last one. Also, after it finishes catching, it sometimes clicks on the screen and the character starts running around, which gives away that the movement is robotic. Maybe it's just happening to me.

**Comment by Laur1810** [comment]
Ah, must also depend on the resolution that you use and so on, as it can change the fish window size, glad that you found it out.
Regarding the success rate, I also noticed that it misses sometimes, but I don't think that is something that can be fixed easily, the bot only clicks a few pixels in front of the fish so if the fish makes a sudden turn right when the bot takes the ss to figure where the fish is and to click where it will be, it will miss, this can only be improved I think by finding the perfect window size, but I don't think it can be perfect.
For the times when the character starts running around, it never happened to me, but this must be because of the actions delay, I would suggest increasing the timer of the bot a bit to solve this, even if it would have a break after each minigame

**Comment by vncsms** [comment]
> @Laur1810 managed to make it work in the end, but I had to set FISH_WINDOW_SIZE = (350, 340) it seems. However, I noticed some issues. Now, I'm not sure if it's just me because the fish window size isn't perfectly configured, but on average, out of 10 mini-games, the hack has a success rate of 60% in catching those 3 fish. The rest of the time, it only catches 2 fish and misses the last one. Also, after it finishes catching, it sometimes clicks on the screen and the character starts running around, which gives away that the movement is robotic. Maybe it's just happening to me.

The success rate can be affected by your computer GPU capacity.  It's necessary to have a good GPU, but a poor one can struggle the bot a litle bit. There are people running this bot pretty well in laptops.

**Comment by ghost** [comment]
@vncsms I have a gaming laptop, but I've been using it in silent and battery-saving mode. I'm going to switch it to performance and turbo mode, and I'll come back with updates >)))

**Comment by vncsms** [comment]
@revertl0l It's not necessary. Is just because some people had complaining about that same problem, but their computer are very cheap. Like 2  GBRAM and intel pentium.

**Comment by ghost** [comment]
@vncsms  to be honest i saw that mini windows that appears in the background, and i have 20-30 fps, maybe if i will change the performance, it  will work more smother.

**Comment by vncsms** [comment]
Mine is 30 - 60

**Comment by ghost** [comment]
So with the new update the hack doesn t work, that is the error.
![1BA3A87E-A0FC-45F1-B9FA-2FE77250A27B](https://github.com/vncsms/Metin2FishBot/assets/135658573/18b631e0-75a7-4b3d-b9f0-28059511747d)
Any idea ?

**Comment by vncsms** [comment]
@revertl0l  Can you open a new issue? This one is getting long.
